playground/optimize-deps/__tests__/optimize-deps.spec.ts

```diff
@@ -141,7 +141,7 @@ test('vue + vuex', async () => {
 
 // When we use the Rollup CommonJS plugin instead of esbuild prebundling,
 // the esbuild plugins won't apply to dependencies
-test('esbuild-plugin', async () => {
+test.runIf(isServe)('esbuild-plugin', async () => {
```

Comment:
I really appreciate your work, but I have some feedback I'd like to share. There are some custom esbuild plugins in my app to handle special files in dependencies, as described in [docs](https://vitejs.dev/guide/dep-pre-bundling.html#customizing-the-behavior). If the pre-build feature is removed at build time, is there any other way to keep the custom esbuild plugin consistent in build mode?

Reply:
You'll need to have the same custom handling implemented as a rollup plugin. It would be good if you link to the way you are using it for reference, but I think it would be hard to justify the complexity this PR is removing as it can be still make to work. And later on if we move to Rolldown instead of esbuild, then you will get the consistency back for dependency handling.